### PR TITLE
Wheels: One artefact file per config

### PR DESCRIPTION
**Pull request by danthe3rd**
Also makes it run every week

cc @AbdBarho let me know if this looks good to you

### CODE REVIEW

**Comment by AbdBarho** [comment]
@danthe3rd look good! glad to see it the wheels being put to good use.

File: .github/workflows/wheels.yml [hunk 1]
```diff
@@ -108,16 +114,22 @@ jobs:
...
 
+      - name: Setup proper pytorch dependency in "requirements.txt"
+        run: >
+          sed -i '/pytorch/d' ./requirements.txt
+          echo "pytorch == ${{ matrix.config.torch_version_constraint }}" >> ./requirements.txt
+          cat ./requirements.txt
+
       - name: Install build dependencies
         run: >
           $PY -m pip install wheel setuptools torch==${{ matrix.config.torch_version }} -r requirements.txt --extra-index-url ${{ matrix.config.pip_index }} &&
```
Comment (AbdBarho) [inline .github/workflows/wheels.yml:125]: I wonder if `torch==${{ matrix.config.torch_version }}` is still necessary after doing the new update, would it make sense to just write the version with cuda in the requirements file and remove the torch version from here?

Comment (danthe3rd) [reply .github/workflows/wheels.yml:125]: Good point - will update

File: .github/workflows/wheels.yml [hunk 2]
```diff
@@ -33,17 +37,20 @@ jobs:
...
 
           - torch_version: "1.13.0+cu116"
+            torch_version_constraint: "1.13"
             cuda_version: "11.6.2"
             pip_index: https://download.pytorch.org/whl/cu116
             cuda_run_file: https://developer.download.nvidia.com/compute/cuda/11.6.2/local_installers/cuda_11.6.2_510.47.03_linux.run
 
           # torch 1.12 is built with cuda 11.3 and 11.6, we ignore 11.3 for now
           - torch_version: "1.12.1+cu116"
+            torch_version_constraint: "1.12"
```
Comment (fmassa) [inline .github/workflows/wheels.yml:53]: nit: is this constraint in line with what is being used by `torch_version` just above? (I see `1.12.1`).

Also, wouldn't it be a bit more robust to make the `torch_version` be a function of `torch_version_constraint` + `cuda_version` ?

File: .github/workflows/wheels.yml [hunk 3]
```diff
@@ -108,16 +115,22 @@ jobs:
           submodules: recursive
           path: "."
 
+      - name: Setup proper pytorch dependency in "requirements.txt"
+        run: >
+          sed -i '/pytorch/d' ./requirements.txt
+          echo "pytorch == ${{ matrix.config.torch_version_constraint }}" >> ./requirements.txt
+          cat ./requirements.txt
```
[the remark this answers is not in the document]
Comment (danthe3rd) [reply .github/workflows/wheels.yml:122]: Fixed it - thanks :)

**Pull request merged**